src/library_pthread.js

```diff
@@ -372,7 +372,12 @@ var LibraryPThread = {
         // it could load up the same file. In that case, developer must either deliver the Blob
         // object in Module['mainScriptUrlOrBlob'], or a URL to it, so that pthread Workers can
         // independently load up the same main application file.
-        'urlOrBlob': Module['mainScriptUrlOrBlob']
+        'urlOrBlob':
+#if expectToReceiveOnModule('mainScriptUrlOrBlob')
+        Module['mainScriptUrlOrBlob']
+#else
+        undefined
+#endif
```

Comment:
Can we just skip the `urlOrBlob` when we don't expect to receive `mainScriptUrlOrBlob`.  i.e. can you wrap the member declaration along with the whole comment above in he `if/endif`?